### ISSUE TITLE
websocket - response code: 403

### ISSUE THREAD

**Issue by ecadagiani** [antoniomika/sish#10]
Hello,
I use your server to expose my socket.io backend. I noticed that the Websocket communications do not work.
I have this error: 
```
WebSocket connection to 'wss://backend.tunnel.hellomybot.io/socket.io/?EIO=3&transport=websocket&sid=5bfX1h9MO5yyIxkpAADU' failed: Error during WebSocket handshake: Unexpected response code: 403
```
Smartly socket.io uses another tech. So it's not blocking but it's problematic.

You can test this, with this client code:
```
const socket = io.connect("http://foo.ssi.sh", {transports: ["websocket"]});
```

**Comment by antoniomika** [comment]
Hmm, I tried to reproduce this error using the socket.io example chat server running in a docker container and then exposing that service using sish. To set this up, here are the steps I followed:
```
git clone https://github.com/socketio/socket.io /tmp/socket.io
cd /tmp/socket.io/examples/chat
echo "FROM node:onbuild\nRUN npm install socket.io\n RUN sed -i \"s/require('..\/..')/require('socket.io')/\" index.js\nCMD node index.js" > Dockerfile
docker build -t test .
docker run -itd --rm -p 3000:3000 test
ssh -R 80:localhost:3000 ssi.sh
```
Then I accessed the service at the address that sish outputted (both http/https) and verified that the network monitor was connected to the websocket and was using that as transit for socket.io

Is the issue occurring elsewhere that this example wouldn't show?

I'm gonna close this issue for now. Let me know if there are any other steps I can follow to reproduce the issue. I want to make sure this works for everyone!

**Comment by ecadagiani** [comment]
Hello,
Thank you for taking the time to look.
It's normal that socket.io works, because when websocket don't work, socket.io use pulling (it seems to me).

You can force websocket, here is a method to test:
An server throught SISH (you can use my image, [ecadagiani/socketio-server-helloworld](https://hub.docker.com/r/ecadagiani/socketio-server-helloworld) )

An client, i use this web tools: [https://xipotera.github.io/socket-io-client-dev-tools/](https://xipotera.github.io/socket-io-client-dev-tools/)
In this tools you set your server adresse (throught SISH), and socket option :
```
{"transports": ["websocket"]}
```

And check your browser console ;)

**Comment by antoniomika** [comment]
Cool, now that I understand what you're doing better I figured it out. You're accessing the websocket from a different `origin` than the host it is on. This ultimately causes sish to close the connection with a 403 because that generally means something shady is going on with the upgrade. In 75de6de, I've added the ability to turn off this functionality by setting `-sish.verifyorigin=false` when running sish. This is not disabled by default however, as it could lead to other security issues.
